### PR TITLE
CF Manifest property override

### LINKED ISSUE #664: Add generic-template for CF Manifest based deployments
child of #627

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add cf template as a separate template file to be used in CF manifest based packages
 - Add Map<String, Object> as the property for CloudFoundryApplicationSpec
 - Support ability to pass the CF application manifest as properties for install/upgrade
   - Support CF application manifest properties
 - Refactor existing codebase
 - Update existing packages

Resolves #664

### CODE REVIEW

File: spring-cloud-skipper-platform-cloudfoundry/src/main/java/org/springframework/cloud/skipper/deployer/cloudfoundry/CloudFoundryReleaseManager.java [hunk 1]
```diff
@@ -42,9 +36,13 @@
 import org.springframework.cloud.skipper.server.repository.AppDeployerDataRepository;
 import org.springframework.cloud.skipper.server.repository.ReleaseRepository;
 import org.springframework.cloud.skipper.server.util.ArgumentSanitizer;
-import org.springframework.cloud.skipper.server.util.ManifestUtils;
 import org.springframework.util.Assert;
 
+import static org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryManifestApplicationDeployer.PUSH_REQUEST_TIMEOUT;
+import static org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryManifestApplicationDeployer.STAGING_TIMEOUT;
+import static org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryManifestApplicationDeployer.STARTUP_TIMEOUT;
+import static org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryManifestApplicationDeployer.isNotFoundError;
```
Comment (jvalkeal) [inline spring-cloud-skipper-platform-cloudfoundry/src/main/java/org/springframework/cloud/skipper/deployer/cloudfoundry/CloudFoundryReleaseManager.java]: Maybe to keep same format as others aka `XXX_XXX_XXX`?

Comment (jvalkeal) [reply spring-cloud-skipper-platform-cloudfoundry/src/main/java/org/springframework/cloud/skipper/deployer/cloudfoundry/CloudFoundryReleaseManager.java]: Ah it was like that already

File: spring-cloud-skipper-platform-cloudfoundry/src/main/java/org/springframework/cloud/skipper/deployer/cloudfoundry/CFApplicationManifestUtils.java [hunk 2]
```diff
@@ -87,6 +93,83 @@ public static ApplicationManifest updateApplicationName(Release release) {
 		return updateApplicationName(cfManifestYamlString, release);
 	}
 
+	public static ApplicationManifest updateApplicationManifest(ApplicationManifest cfApplicationManifest,
+			Map<String, Object> manifest) {
+		ApplicationManifest.Builder applicationManifestBuilder = ApplicationManifest.builder()
+				.from(cfApplicationManifest);
+		for (Map.Entry<String, Object> manifestEntry: manifest.entrySet()) {
+			if (manifestEntry.getKey().equalsIgnoreCase("buildpack")) {
```
Comment (jvalkeal) [inline spring-cloud-skipper-platform-cloudfoundry/src/main/java/org/springframework/cloud/skipper/deployer/cloudfoundry/CFApplicationManifestUtils.java]: Should we extract all these to shared static variables?

File: spring-cloud-skipper-server-core/src/test/resources/repositories/sources/test/python/python-printer-1.0.0/values.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Default values for {{name}}
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates
+spec:
+  resource: https://github.com/ilayaperumalg/sandbox/raw/master/python/hello.py.zip
```
Comment (jvalkeal) [inline spring-cloud-skipper-server-core/src/test/resources/repositories/sources/test/python/python-printer-1.0.0/values.yml]: Can we stop using your git repo for these?

Comment (ilayaperumalg) [reply spring-cloud-skipper-server-core/src/test/resources/repositories/sources/test/python/python-printer-1.0.0/values.yml]: yeah, this is just a placeholder for the non-java applications. What do you think as a better alternative?

Comment (jvalkeal) [reply spring-cloud-skipper-server-core/src/test/resources/repositories/sources/test/python/python-printer-1.0.0/values.yml]: Either in repo itself or just attaching files to a dummy gh issue(in other projects I've just created a dummy issue, close it and then you can just drag and drop files there and github creates a link).

File: spring-cloud-skipper/src/main/resources/org/springframework/cloud/skipper/io/cf-template.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: skipper.spring.io/v1
```
Comment (jvalkeal) [inline spring-cloud-skipper/src/main/resources/org/springframework/cloud/skipper/io/cf-template.yml]: Btw, where this template is used? Can't find references to it as I can with `generic-template.yml`.

**Comment by jvalkeal** [comment]
ok, lgtm for now, merging

**Pull request closed without merging**